### ISSUE TITLE
Git clone fails halfway through

### ISSUE THREAD

**Issue by WallE256** [keybase/client#9015]
Hello,
I have moved a ~600MB repository with my Mac with success, however, when I try to clone that same repository on my Windows machine, the `git clone` stops halfway through with no apparent reason. The only thing that I noticed might be a problem is a rampant memory usage by the `git-remote` module. Here is how the console looks like after a failed clone:
```
C:\Users\WallE256>git clone keybase://team/ichb_ftc_dev/robovamps_app
Cloning into 'robovamps_app'...
Initializing Keybase... done.
Syncing with Keybase... done.
Counting: 599.43 MB... done.
Cryptographic cloning: (47.31%) 283.57/599.43 MB...
C:\Users\WallE256>
```

Let me know if I can provide any additional information.
Thanks.

my log id: b9d9b53eb2ca0c0c01d9e71c

**Comment by strib** [comment]
@WallE256 thanks for the report.  I suspect this is due to the fact that we do 32-bit builds for Windows, even for 64-bit architectures, and that leads to earlier memory issues that we wouldn't see otherwise.

What do you mean by "rampant memory usage"?  How much was it using before the crash?

**Comment by WallE256** [comment]
When the clone starts, it starts from about 40MB memory usage, and it crashes when it reaches 1GB memory usage. The actual memory usage slightly differs from test to test (crashes occur from 800MB to about 1GB memory usage), but the behavior is the same. So basically by "rampant memory usage" I'm talking about a probable memory leak.

**Comment by strib** [comment]
Hmm, ok thanks for the info.  How much RAM does that machine/VM have in total?

We heavily cache data in our process, and so reaching 800 MB - 1GB for such a big repo doesn't necessarily mean a _leak_ (although there could definitely be one).  It might just mean we're not sizing our caches correctly.

**Comment by WallE256** [comment]
I have 16 GB of RAM on my computer, so crashing from actually running out of memory are out of the question.

**Comment by strib** [comment]
Yeah I think the memory limits are artificially low because it's a 32-bit build.

Thanks for the info, we're looking into it.

**Comment by strib** [comment]
So it turns out we've already fixed the memory issues in master, and they will become available with the next Windows release (likely next week).  In theory you could build it yourself from our github repo, if you're so inclined.  Sorry for the inconvenience in the meantime!
